### PR TITLE
GitHub module style

### LINKED ISSUE #100: Update module structure
Adopt module structure to the [Github internal fork](https://github.com/github/puppet-consul_template). It has several advantages:

- The interfaces provide a config_hash and config_defaults instead of individual parameters to both allow access to the full set of configuration options and avoid the need to make module changes as consul-template configuration evolves.

- It allows a much leaner implementation, adding options to consul_template::config is quite confusing.

### PR DESCRIPTION
**Pull request by sts**
Adopt module structure to the [Github internal fork](https://github.com/github/puppet-consul_template). Closes #100 

The code was tested on Debian so far and worked, also added a more detailed changelog entry to outline the changes.

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
   "dependencies": [
     { "name": "puppetlabs/stdlib", "version_requirement": ">=1.0.0 < 5.0.0" },
     { "name": "puppetlabs/concat", "version_requirement": ">= 2.0.0 < 3.0.0" },
-    { "name": "puppet/staging", "version_requirement": ">= 0.4.0 <= 2.2.0" }
+    { "name": "puppet/staging", "version_requirement": ">= 0.4.0 <= 2.2.0" },
+    { "name": "solarkennedy/puppet-consul", "version_requirement": ">= 2.1.1" }
```
Comment (craigwatson) [inline metadata.json]: This should be `KyleAnderson/consul`

**Review by craigwatson — CHANGES_REQUESTED** [review]
Thanks for all your work, looks great!

There's one minor metadata change here which should allow tests to pass :)

File: spec/classes/consul_template_spec.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@
       context "on #{os}" do
         let(:facts) do
           facts.merge({
-            'kernel'           => 'Linux',
+            #'kernel'           => 'Linux',
```
Comment (wyardley) [inline spec/classes/consul_template_spec.rb]: Probably should remove this vs. commenting it out?

File: spec/defines/watch_spec.rb [hunk 3]
```diff
@@ -4,60 +4,42 @@
 describe 'consul_template::watch', :type => :define do
   context 'supported operating systems' do
     on_supported_os.each do |os, facts|
-
       context "on #{os}" do
         let(:facts) do
           facts.merge({
             :kernel => 'Linux',
-            :staging_http_get => 'curl'
+            :operatingsystem => os,
```
Comment (wyardley) [inline spec/defines/watch_spec.rb]: Shouldn't `:kernel` and `:operatingsystem` already  be available from rspec-puppet-facts?

**Comment by wyardley** [comment]
@sts Thanks for doing this, much appreciated, as we rely on this functionality as well (though rebasing #100 is going to be a lot of fun).
👍
Put a couple comments inline, though  I don't think they're super important.

File: manifests/watch.pp [hunk 4]
```diff
@@ -4,54 +4,59 @@
...
-  if count([$content, $source, $template]) > 1 {
-    err('Specify either content, source or template parameter for consul_template::watch, but only a single parameter should be specified')
+  if $template != undef and $config_hash_real['source'] != undef {
+    err ('Specify either template parameter or config_hash["source"] for consul_template::watch - but not both')
   }
 
-  if $source != undef {
-    $source_name = $source
-    $frag_name   = $source
+  if $template == undef {
```
Comment (wyardley) [inline manifests/watch.pp]: Maybe `unless $template` accomplishes the same thing in a more idiomatic way?

**Comment by wyardley** [comment]
@ross FYI, this attempts to bring the changes from your fork back upstream.
Also proposing some major changes in #108 (not yet merged as of time of writing). Would love if you have any feedback.

**Pull request merged**